src/software/jetson_nano/services/motor.cpp

```diff
@@ -102,54 +111,223 @@ MotorService::MotorService(const RobotConstants_t& robot_constants,
     OPEN_SPI_FILE_DESCRIPTOR(front_right, FRONT_RIGHT_MOTOR_CHIP_SELECT)
     OPEN_SPI_FILE_DESCRIPTOR(back_left, BACK_LEFT_MOTOR_CHIP_SELECT)
     OPEN_SPI_FILE_DESCRIPTOR(back_right, BACK_RIGHT_MOTOR_CHIP_SELECT)
-    OPEN_SPI_FILE_DESCRIPTOR(dribbler, DRIBBLER_MOTOR_CHIP_SELECT)
+    // TODO (#2606) - Add Dribbler Motor support
 
     // Make this instance available to the static functions above
     g_motor_service = this;
+
+    // Clear faults
+    reset_gpio.setValue(GpioState::LOW);
+    sleep(1);
+    reset_gpio.setValue(GpioState::HIGH);
+    sleep(1);
+
+    // TMC6100 Setup
+    startDriver(FRONT_LEFT_MOTOR_CHIP_SELECT);
+    startDriver(BACK_RIGHT_MOTOR_CHIP_SELECT);
+    startDriver(FRONT_RIGHT_MOTOR_CHIP_SELECT);
+    startDriver(BACK_LEFT_MOTOR_CHIP_SELECT);
+
+    // Check faults
+    checkDriverFault(FRONT_LEFT_MOTOR_CHIP_SELECT);
+    checkDriverFault(FRONT_RIGHT_MOTOR_CHIP_SELECT);
+    checkDriverFault(BACK_LEFT_MOTOR_CHIP_SELECT);
+    checkDriverFault(BACK_RIGHT_MOTOR_CHIP_SELECT);
+
+    // TMC4671 Setup
+    startController(FRONT_LEFT_MOTOR_CHIP_SELECT);
+    startController(BACK_RIGHT_MOTOR_CHIP_SELECT);
+    startController(FRONT_RIGHT_MOTOR_CHIP_SELECT);
+    startController(BACK_LEFT_MOTOR_CHIP_SELECT);
```

Comment:
perhaps we can loop over these e.g. for each motor m, startDriver(m), for each motor m, checkDriverFault(m) etc. 
Just noticing that they are in different orders so we might have a hard to see bug

Reply:
good call, fixed!